pallets/cash/src/internal/set_yield_next.rs

```diff
@@ -15,48 +15,38 @@ use types_derive::Types;
 
 #[derive(Copy, Clone, Eq, PartialEq, Encode, Decode, Debuggable, Types)]
 pub enum SetYieldNextError {
-    TimestampTooSoonToNow,
-    TimestampTooSoonToNext,
+    NotEnoughTimeToSyncBeforeNext,
+    NotEnoughTimeToSyncBeforeCancel,
 }
 
-fn get_cash_yield_index_after<T: Config>(change_in_time: Timestamp) -> Result<CashIndex, Reason> {
+fn get_cash_yield_index_after<T: Config>(dt: Timestamp) -> Result<CashIndex, Reason> {
     let cash_yield = CashYield::get();
     let cash_index_old = GlobalCashIndex::get();
-    let increment = cash_yield.compound(change_in_time)?;
-    cash_index_old
-        .increment(increment.into())
-        .map_err(Reason::MathError)
+    let increment = cash_yield.compound(dt)?;
+    Ok(cash_index_old.increment(increment.into())?)
 }
 
 pub fn set_yield_next<T: Config>(
     next_yield: APR,
     next_yield_start: Timestamp,
 ) -> Result<(), Reason> {
-    let now = get_now::<T>();
-    let change_in_time = next_yield_start
-        .checked_sub(now)
-        .ok_or(Reason::TimeTravelNotAllowed)?;
+    let now = get_recent_timestamp::<T>()?;
+    let min_t = now + MIN_NEXT_SYNC_TIME;
 
-    // TODO: Maybe check if `now` is zero / default?
+    require!(
+        next_yield_start >= min_t,
+        SetYieldNextError::NotEnoughTimeToSyncBeforeNext.into()
+    );
 
-    // Read Require CashYieldNext.NextAPRStartNow+ParamsMinNextSyncTime
     if let Some((_, next_start)) = CashYieldNext::get() {
         require!(
-            next_start >= now + MIN_NEXT_SYNC_TIME,
-            Reason::SetYieldNextError(SetYieldNextError::TimestampTooSoonToNext)
+            next_start >= min_t,
+            SetYieldNextError::NotEnoughTimeToSyncBeforeCancel.into()
         );
     }
 
-    // Note: Any already pending change cannot be canceled unless the min sync time is also met for the cancel.
-    // Read Require NextAPRStartNow+ParamsMinNextSyncTime
-    require!(
-        next_yield_start >= now + MIN_NEXT_SYNC_TIME,
-        Reason::SetYieldNextError(SetYieldNextError::TimestampTooSoonToNow)
-    );
-
-    // XXX fix this up
-    // Read NextYieldIndex=GetCashYieldIndexAt(NextAPRStart)
-    let next_yield_index: CashIndex = get_cash_yield_index_after::<T>(change_in_time)?;
+    let dt = next_yield_start - now;
```

Comment:
Same with the unchecked sub.

Reply:
This one is safe I think since we check `next_yield_start >= now + C`  where `C > 0` above

Reply:
Though, with a potential wrap around.... I wonder based on every other inefficiency and the fact this already returns a `Result` if we should just do the check